source/unified-test-format/unified-test-format.rst

```diff
@@ -0,0 +1,1896 @@
+omitted (or otherwise filtered out) from the list of observed command monitoring
+events for this client.
+
+An example of this operation follows::
+
+    # Enable the fail point on the mongos to which session0 is pinned
+    - name: targetedFailPoint
+      object: testRunner
+      arguments:
+        session: *session0
+        failPoint:
+          configureFailPoint: failCommand
+          mode: { times: 1 }
+          data:
+            failCommands: ["commitTransaction"]
+            closeConnection: true
+
+**TODO**: `allowMultipleMongoses`_ only affects the test runner if it is false
+(i.e. a connection string to a sharded cluster is reduce to one mongos). The
+transaction spec hinted that ``useMultipleMongoses: true`` would force a
+connection string to contain multiple mongos servers, but that seems untenable
+since single-mongos and standalone connection strings are indistinguishable from
+one another. Should we change `allowMultipleMongoses`_ to actively require
+multiple mongos servers if true (omitted would still be a no-op)? That may be
+possible if we defer enforcement until after the topology is determined at
+runtime.
+
+
+assertSessionTransactionState
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``assertSessionTransactionState`` operation instructs the test runner to
+assert that the transaction state of the given session is equal to the specified
+value. The possible values are as follows: ``none``, ``starting``,
+``in_progress``, ``committed``, ``aborted``.
+
+An example of this operation follows::
+
+    - name: assertSessionTransactionState
+      object: testRunner
+      arguments:
+        session: *session0
+        state: in_progress
+
+
+assertSessionPinned
+~~~~~~~~~~~~~~~~~~~
+
+The ``assertSessionPinned`` operation instructs the test runner to assert that
+the given session is pinned to a mongos.
+
+An example of this operation follows::
+
+    - name: assertSessionPinned
+      object: testRunner
+      arguments:
+        session: *session0
+
+
+assertSessionUnpinned
+~~~~~~~~~~~~~~~~~~~~~
+
+The ``assertSessionUnpinned`` operation instructs the test runner to assert that
+the given session is not pinned to a mongos.
+
+An example of this operation follows::
+
+    - name: assertSessionPinned
+      object: testRunner
+      arguments:
+        session: *session0
+
+
+assertCollectionExists
+~~~~~~~~~~~~~~~~~~~~~~
+
+The ``assertCollectionExists`` operation instructs the test runner to assert
+that the given collection exists in the database.
+
+An example of this operation follows::
+
+    - name: assertCollectionExists
+      object: testRunner
+      arguments:
+        database: db
+        collection: test
+
+Use a ``listCollections`` command to check whether the collection exists. Note
+that it is currently not possible to run ``listCollections`` from within a
+transaction.
+
+**TODO**: If this will refer to a collection entity, database is redundant.
```

Comment:
I think `collection` should be an entity ID for consistency. Anything that is in the entity map should always be referred to by its ID.

Reply:
Special Test Operations other than `targetedFailPoint` should all use the internal MongoClient, although I realize I only specified that for `failPoint`. IMO, this is like a `test.outcome` assertion.

Should we keep this as-is provided I document that all of these assertions use the internal MongoClient?

Although this mainly exists for transactions, while writing this I was thinking about how `assertCollectionExists` might be used with an aggregation output collection, which I wouldn't expect to ever define as an entity.

Reply:
Will rename these to `databaseName` and `collectionName` for consistency.

Reply:
`databaseName` and `collectionName` are now documented in Common Options and I updated all arguments to refer to those definitions.